tests/ecn/files/helper.py

```diff
@@ -0,0 +1,345 @@
+        dst=FieldPattern(choice='01:80:C2:00:00:01'),
+        src=FieldPattern(choice='00:00:fa:ce:fa:ce'),
+        class_enable_vector=FieldPattern(choice=vector),
+        pause_class_0=FieldPattern(choice=pause_time[0]),
+        pause_class_1=FieldPattern(choice=pause_time[1]),
+        pause_class_2=FieldPattern(choice=pause_time[2]),
+        pause_class_3=FieldPattern(choice=pause_time[3]),
+        pause_class_4=FieldPattern(choice=pause_time[4]),
+        pause_class_5=FieldPattern(choice=pause_time[5]),
+        pause_class_6=FieldPattern(choice=pause_time[6]),
+        pause_class_7=FieldPattern(choice=pause_time[7]),
+    ))
+
+    """ Pause frames are sent from the RX port """
+    pause_src_point = PortTxRx(tx_port_name=testbed_config.ports[rx_port_id].name,
+                               rx_port_name=testbed_config.ports[tx_port_id].name)
+
+    speed_str = testbed_config.layer1[0].speed
+    speed_gbps = int(speed_str.split('_')[1])
+    pause_dur = 65535 * 64 * 8.0 / (speed_gbps * 1e9)
+    pps = int(2 / pause_dur)
+
+    pause_flow = Flow(
+        name=pause_flow_name,
+        tx_rx=TxRx(pause_src_point),
+        packet=[pause_pkt],
+        size=Size(64),
+        rate=Rate('pps', value=pps),
+        duration=Duration(FixedSeconds(seconds=exp_dur_sec,
+                                       delay=0,
+                                       delay_unit='nanoseconds'))
+    )
+
+    result.append(pause_flow)
+    return result
+
+
+def __config_capture(testbed_config, port_id):
+    """
+    Generate packet capture configuration
+
+    Args:
+        testbed_config (obj): L2/L3 config of a T0 testbed
+        port_id (int): ID of DUT port to capture packets
+
+    Returns:
+        Packet capture configuration (list)
+    """
+
+    """ We only capture IP packets """
+    ip_filter = CustomFilter(filter='40', mask='0f', offset=14)
+    result = [Capture(name='rx_capture',
+                      port_names=[testbed_config.ports[port_id].name],
+                      choice=[BasicFilter(ip_filter)],
+                      enable=True)]
+    return result
+
+def __run_traffic(api,
+                  config,
+                  all_flow_names,
+                  exp_dur_sec,
+                  capture_port_name,
+                  pcap_file_name):
+    """
+    Run traffic and capture packets
+
+    Args:
+        api (obj): IXIA session
+        config (obj): experiment config
+        all_flow_names (list): names of all the flows
+        capture_port_name (str): name of the port to capture packets
+        pcap_file_name (str): name of the pcap file to store captured packets
+
+    Returns:
+        N/A
+    """
+    api.set_state(State(ConfigState(config=config, state='set')))
+
+    api.set_state(State(PortCaptureState(port_names=[capture_port_name],
+                                         state='start')))
+
+    api.set_state(State(FlowTransmitState(state='start')))
+    time.sleep(exp_dur_sec)
+
+    while True:
```

Comment:
instead of indefinite loop, can we also keep a cnt to stop polling after max number of cnts. In case, there is any issue on the tgen side, we will never break out of this loop

Reply:
Fixed